python/pyspark/pandas/series.py

```diff
@@ -4548,22 +4548,31 @@ def update(self, other: "Series") -> None:
         if not isinstance(other, Series):
             raise TypeError("'other' must be a Series")
 
-        combined = combine_frames(self._psdf, other._psdf, how="leftouter")
+        if same_anchor(self, other):
+            scol = F.when(other.spark.column.isNotNull(), other.spark.column).otherwise(
+                self.spark.column
+            )
```

Comment:
We should alias the `scol` with the original name?

`scol = ... .alias(self._internal.data_spark_column_names[0])`

Reply:
thanks , updated